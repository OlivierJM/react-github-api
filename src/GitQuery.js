import React from "react";
import gql from "graphql-tag";
import { Query } from "react-apollo";

const GET_REPOS = gql`
  query GETREPOS($login: String!) {
    user(login: $login) {
      repositoriesContributedTo(last: 5) {
        totalCount
        nodes {
          name
        }
      }
      repositories(last: 50) {
        totalCount
        nodes {
          name
          url
          description
          stargazers(last: 5) {
            nodes {
              name
            }
          }
        }
      }
    }
  }
`;

export const Repos = ({ login }) => (
  <Query query={GET_REPOS} variables={{ login }} fetchPolicy={"network-only"}>
    {
      ({ loading, data }) =>
      (!loading && data && (
        <ol>
          {data.user.repositories.nodes.map((repo, i) => (
            <div key={i} className="card" style={{width: '20em'}}>
            <div className="card-body">
              <h4 className="card-title">{repo.name}</h4>
              <h5 className="card-subtitle">Nice looking subtitle.</h5>
              <p className="card-text">This is another example of a card without image. Cards are also meant to be used without images, but with text/links/buttons.</p>
              <a className="card-link" href={repo.url}>First link</a>
            </div>
            </div>

          ))}
        </ol>
      )) || <p>Loading...</p>
    }
  </Query>
);
