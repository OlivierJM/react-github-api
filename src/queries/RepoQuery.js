import gql from "graphql-tag"

const GET_REPOS = gql`
  query GETREPOS($login: String!) {
    user(login: $login) {
      name
      organizations {
        totalCount
      }
      avatarUrl
      location
      repositories(
        first: 100
        orderBy: { field: STARGAZERS, direction: DESC }
      ) {
        totalCount
        nodes {
          name
          url
          description
          isFork
          stargazers(last: 5) {
            totalCount
          }
        }
      }
    }
  }
`

export { GET_REPOS }
