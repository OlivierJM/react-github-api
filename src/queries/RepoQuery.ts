import gql from "graphql-tag"

const GET_REPOS = gql`
  query GETREPOS($login: String!) {
    user(login: $login) {
      name
      organizations {
        totalCount
      }
      # this will be used on the new page for user info
      contributionsCollection {
        totalCommitContributions
        totalIssueContributions
        totalPullRequestContributions
        totalRepositoriesWithContributedIssues
        totalRepositoriesWithContributedPullRequestReviews
        totalRepositoryContributions
        contributionCalendar {
          totalContributions
        }
      }
      # user image
      avatarUrl
      location
      repositories(
        first: 100
        orderBy: { field: STARGAZERS, direction: DESC }
      ) {
        totalCount
        nodes {
          name
          url
          description
          isFork
          stargazers(last: 5) {
            totalCount
          }
        }
      }
    }
  }
`

export { GET_REPOS }
