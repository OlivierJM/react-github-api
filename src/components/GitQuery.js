import React from "react"
import { Query } from "react-apollo"
import { GET_REPOS } from "../queries/RepoQuery"
import PlaceHolder from "./PlaceHolder"

export const Repos = ({ login }) => (
  <Query query={GET_REPOS} variables={{ login }} fetchPolicy={"network-only"}>
    {({ loading, data }) =>
      !loading && data ? (
        data.user.repositories.nodes.map((repo, i) => (
          <div className="sm-12 md-4 col" key={i}>
            <div className="card">
              <div className="card-body">
                <h4 className="card-title">{repo.name}</h4>
                <h5 className="card-subtitle">
                  <span role="img" aria-label="stars">
                    ‚≠ê
                  </span>{" "}
                  {repo.stargazers.totalCount}
                </h5>
                <h5 className="card-subtitle">{repo.isFork && "Forked"}</h5>
                <p className="card-text">{repo.description}</p>
                <a
                  className="card-link"
                  target="_blank"
                  rel="noopener noreferrer"
                  href={repo.url}
                >
                  Github Link
                </a>
              </div>
            </div>
          </div>
        ))
      ) : !login.length ? (
        <PlaceHolder />
      ) : (
        <div className="row flex-center">
          <p className="flex-center">Loading ....</p>
        </div>
      )
    }
  </Query>
)
