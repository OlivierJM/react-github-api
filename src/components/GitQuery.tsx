import * as React from "react"
import { useQuery } from "react-apollo-hooks"
import { GET_REPOS } from "../queries/RepoQuery"
import PlaceHolder from "./PlaceHolder"
import Profile from "./Profile"
import "./styles/gitquery.css"

interface repoProps {
  login: string
}

interface repo {
  name: string
  stargazers: stargazers
  isFork: boolean
  description: string
  url: string
}
interface stargazers {
  totalCount: number
}

const Repos: React.SFC<repoProps> = ({ login }) => {
  const { data, error, loading } = useQuery(GET_REPOS, {
    variables: { login },
    suspend: true,
  })
  // fix the returned component to be different
  if (loading) {
    return <PlaceHolder />
  }
  // The query will originally contain errors because of the initial empty string we are passing üôà
  if (error) {
    return <PlaceHolder />
  }
  return (
    <React.Fragment>
      <div className="row">
        <Profile
          totalCount={data.user.repositories.totalCount}
          name={data.user.name}
          orgs={data.user.organizations.totalCount}
          location={
            data.user.contributionsCollection.contributionCalendar
              .totalContributions
          }
        />
      </div>
      <div className="container">
        {data.user.repositories.nodes.map((repo: repo, i: number) => (
          <div className="item" key={i}>
            <div className="">
              <div className="">
                <h4 className="">{repo.name}</h4>
                <h5 className="">
                  <span role="img" aria-label="stars">
                    ‚≠ê
                  </span>{" "}
                  {repo.stargazers.totalCount}
                </h5>
                <h5 className="">{repo.isFork && "Forked"}</h5>
                <p className="">{repo.description}</p>
                <a
                  className=""
                  target="_blank"
                  rel="noopener noreferrer"
                  href={repo.url}
                >
                  Github Link
                </a>
              </div>
            </div>
          </div>
        ))}
      </div>
    </React.Fragment>
  )
}

export { Repos }
