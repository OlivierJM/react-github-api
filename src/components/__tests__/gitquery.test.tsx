import React from "react"
import { shallow } from "enzyme"
import { MockedProvider } from "react-apollo/test-utils"
import { Repos } from "../GitQuery"
import { GET_REPOS } from "../../queries/RepoQuery"

const mocks = [
  {
    request: {
      query: GET_REPOS,
      variables: {
        name: "olivier",
      },
    },
    result: {
      data: {
        user: { id: "1", name: "olivier" },
      },
    },
  },
]

describe("The Query component ", () => {
  const wrapper = shallow(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Repos login="olivier" />
    </MockedProvider>
  )
  it("should render and have the proper value", () => {
    expect(wrapper.find({ login: "olivier" }))
  })
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("should have a card class", () => {
    expect(wrapper.exists(".row")).toEqual(true)
    expect(wrapper.find(".card").exists()).toEqual(true)
  })
})
