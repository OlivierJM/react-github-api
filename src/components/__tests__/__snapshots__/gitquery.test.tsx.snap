// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Query component  should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MockedProvider
    addTypename={false}
    mocks={
      Array [
        Object {
          "request": Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GETREPOS",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "login",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "login",
                              },
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "user",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "organizations",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalCount",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "contributionsCollection",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalCommitContributions",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalIssueContributions",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalPullRequestContributions",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalRepositoriesWithContributedIssues",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalRepositoriesWithContributedPullRequestReviews",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalRepositoryContributions",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "contributionCalendar",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "totalContributions",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "avatarUrl",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "location",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "first",
                                  },
                                  "value": Object {
                                    "kind": "IntValue",
                                    "value": "100",
                                  },
                                },
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "orderBy",
                                  },
                                  "value": Object {
                                    "fields": Array [
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "field",
                                        },
                                        "value": Object {
                                          "kind": "EnumValue",
                                          "value": "STARGAZERS",
                                        },
                                      },
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "direction",
                                        },
                                        "value": Object {
                                          "kind": "EnumValue",
                                          "value": "DESC",
                                        },
                                      },
                                    ],
                                    "kind": "ObjectValue",
                                  },
                                },
                              ],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "repositories",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "totalCount",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "nodes",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "name",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "url",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "description",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "isFork",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [
                                            Object {
                                              "kind": "Argument",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "last",
                                              },
                                              "value": Object {
                                                "kind": "IntValue",
                                                "value": "5",
                                              },
                                            },
                                          ],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "stargazers",
                                          },
                                          "selectionSet": Object {
                                            "kind": "SelectionSet",
                                            "selections": Array [
                                              Object {
                                                "alias": undefined,
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "totalCount",
                                                },
                                                "selectionSet": undefined,
                                              },
                                            ],
                                          },
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [
                    Object {
                      "defaultValue": undefined,
                      "directives": Array [],
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "String",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "login",
                        },
                      },
                    },
                  ],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 888,
                "start": 0,
              },
            },
            "variables": Object {
              "name": "olivier",
            },
          },
          "result": Object {
            "data": Object {
              "user": Object {
                "id": "1",
                "name": "olivier",
              },
            },
          },
        },
      ]
    }
  >
    <Repos
      login="olivier"
    />
  </MockedProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Repos
        login="olivier"
      />,
      "client": ApolloClient {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "version": "2.4.12",
        "watchQuery": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "login": "olivier",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Repos
          login="olivier"
        />,
        "client": ApolloClient {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "clientAwareness": Object {},
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "version": "2.4.12",
          "watchQuery": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "login": "olivier",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
